src/dmd/link.d

```diff
@@ -569,38 +569,21 @@ public int runLINK()
             argv.push("-Xlinker");
             argv.push("--gc-sections");
         }
-        /* Add each library, prefixing it with "-l".
+        /* Add each library, prefixing it with "-l" if necessary.
          * The order of libraries passed is:
-         *  1. any libraries passed with -L command line switch
-         *  2. libraries specified on the command line
+         *  1. static libraries ending with *.a
+         *  2. link switches passed with -L command line switch
          *  3. libraries specified by pragma(lib), which were appended
          *     to global.params.libfiles.
-         *  4. link switches, that may also contain -l libraries
-         *  5. standard libraries.
```

Comment:
Assuming this, now changed, documentation was correct (actually matching the implementation) won't this PR introduce some other breaking changes?

Reply:
Not sure this doc was correct (will make this clearer). I am not a `ld` guru, but I don't think other breaking change be introduced.
Actual order has evolved in the following way
before #8172:
```d
        /*  1. link switches passed with -L command line switch (global.params.linkswitches)
         *  2. static libraries (ending with *.a)  (global.params.libfiles)
         *  3. libraries specified by pragma(lib), prefixed with "-l" (global.params.libfiles)
         *  4. dynamic libraries passed to the command line (global.params.dllfiles)
         *  5. standard libraries.
         */
```
after #8172:
```d
        /*  1. static libraries (ending with *.a)  (global.params.libfiles)
         *  2. libraries specified by pragma(lib), prefixed with "-l" (global.params.libfiles)
         *  3. link switches passed with -L command line switch (global.params.linkswitches)
         *  4. dynamic libraries passed to the command line (global.params.dllfiles)
         *  5. standard libraries.
         */
```
With this PR:
```d
        /*  1. static libraries (ending with *.a)  (global.params.libfiles)
         *  2. link switches passed with -L command line switch (global.params.linkswitches)
         *  3. libraries specified by pragma(lib), prefixed with "-l" (global.params.libfiles)
         *  4. dynamic libraries passed to the command line (global.params.dllfiles)
         *  5. standard libraries.
         */
```